lazy_dataset/core.py

```diff
@@ -2275,18 +2275,20 @@ def __iter__(self):
             for j, (bucket, _) in enumerate(buckets):
                 found_bucket = bucket.maybe_append(example)
                 if found_bucket:
-                    if bucket.is_completed():
-                        data = bucket.data
-                        if self.sort_key is not None:
-                            data = sorted(data, key=self.sort_key, reverse=self.reverse_sort)
-                        yield data
-                        buckets.pop(j)
                     break
             if not found_bucket:
-                new_bucket = self.bucket_cls(
+                bucket = self.bucket_cls(
```

Comment:
I think I would do it like this (with better naming)
```python
bucket = None
for j, (bucket_, _) in enumerate(buckets):
    if bucket_.maybe_append(example):
        bucket = bucket_
        break
if bucket is None:
    bucket = self.bucket_cls(...)
```
This removes one variable (`found_bucket`) and doesn't use the (sometimes undefined) loop variable `bucket` after the loop. But, this is just personal taste.